### ISSUE TITLE
Proposal for Example Programs That Illustrate Python APIs

### ISSUE THREAD

**Issue by jimthompson5802** [uber/ludwig#560]
**Is your feature request related to a problem? Please describe.**
N/A

**Describe the use case**
### Proposal
Create example programs that demonstrate Ludwig's Python apis.  Example programs will mirror subset of [Ludwig's command line examples](https://uber.github.io/ludwig/examples/).  This proposal covers Python programs for 
* Titanic Survivor
* MNIST
* Text Classification

Sample programs will perform perform one or more of the following functions:
* Training
* Predictions
* Visualizations
* Accessing and analyzing Ludwig's results files (e.g., training_statistics.json)

[This](https://github.com/jimthompson5802/ludwig_examples/tree/master/mnist) illustrates the proposal.

If desired, example programs can be provided in both native python programs (*.py) and Jupyter notebooks (*.ipynb) formats.

A new sub-directory called `examples` will be created in the project's root directory.  Structure of the `examples` of this new directory follows:
``` 
    examples/
        titanic/
            model_training.py
            model_training.ipynb
            model_definition.yaml
            . . .
        mnist/
            . . .
        text/
            . . .
```

 ### Considerations
 * If major changes are expected in the apis, then creating example programs 
 at this time may be premature. 
 * If significant changes are planned to the structure or format of the results files, then example programs demonstrating access may not make sense.

**Describe the solution you'd like**
Sample programs and instructions for running the programs.

**Describe alternatives you've considered**
N/A

**Additional context**
I'm willing to work on this if the proposal makes sense.

**Comment by msaisumanth** [comment]
Thanks @jimthompson5802. This is a great idea. We had discussed something before.

@w4nderlust @ydudin3 what do you think?

**Comment by jimthompson5802** [comment]
@msaisumanth   If it helps, I can submit a PR that contains example code for for one of the categories, i.e., titanic or mnist.  This way you'll have a concrete example of what I meant.  I can incorporate any feedback on the example code, make sure I'm using the apis as intended by the project's design objectives.  Once we are in agreement, you can merge into master and I'll start on the next set.

**Comment by msaisumanth** [comment]
yeah, that would be really helpful. Thanks!

**Comment by jimthompson5802** [comment]
Closed by #564, #578, #599
